File: internal/web/vo/endpointslice.go
package vo

import (
	discoveryv1 "k8s.io/api/discovery/v1"
	"time"
)

type EndpointSliceVO struct {
	BaseVO
	AddressType string    `json:"addressType"`
	CreateTime  time.Time `json:"createTime"`
	Yaml        string    `json:"yaml"`
}

func EndpointSlice2VO(eps discoveryv1.EndpointSlice) EndpointSliceVO {
	return EndpointSliceVO{
		BaseVO: BaseVO{
			Name:      eps.Name,
			Namespace: eps.Namespace,
		},
		AddressType: string(eps.AddressType),
		CreateTime:  eps.CreationTimestamp.Time,
	}
}

File: front/src/api/endpointslice.ts
import { request } from '@/utils/request'
import type { ApiResponse, PageResponse } from './types'

// EndpointSlice查询参数
export interface EndpointSliceQueryDTO {
  clusterId: string
  namespace?: string
  name?: string
  page?: number
  pageSize?: number
  keyword?: string
}

// EndpointSlice视图对象
export interface EndpointSliceVO {
  name: string
  namespace: string
  addressType: string
  createTime: string
  yaml?: string
}

// EndpointSlice API
export const endpointSliceApi = {
  // 获取EndpointSlice列表
  getEndpointSliceList: (params: EndpointSliceQueryDTO): Promise<ApiResponse<PageResponse<EndpointSliceVO>>> => {
    return request.get('/endpointslice/list', { params })
  }
}